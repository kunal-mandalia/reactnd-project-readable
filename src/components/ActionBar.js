import React from 'react'
import PropTypes from 'prop-types'
import '../styles/ActionBar.css'
import { friendlyDurationSince } from '../helper/functions'

const ActionBar = ({
  id,
  editTitle,
  editBody,
  date,
  author,
  category,
  hideCategory,
  editMode,
  onCancel,
  onDelete,
  onEdit,
  onSave,
  onReply,
}) => {
  const since = friendlyDurationSince(Date.now(), date)
  return (
    <div className='action-bar'>
      <hr className='action-bar-separator'/>
      <span className='action-bar-date'> {since} </span>
      <span className='action-bar-author'> by {author} </span>
      <span className={hideCategory ? 'hidden' : 'action-bar-category'}> in {category} </span>
      {editMode ? (
        <span className='action-bar-actions'>
          [ 
            <a className='action-bar-action' onClick={() => { onSave(id, editTitle, editBody) }}> save </a> | 
            <a className='action-bar-action' onClick={() => { onDelete(id) }}> delete </a> | 
            <a className='action-bar-action' onClick={() => { onCancel(id) }}> cancel </a>
          ]
        </span>
      ) : (
        <span className='action-bar-actions'>
          [
            <a className='action-bar-action' onClick={() => { onEdit(id) }}> edit </a> | 
            <a className='action-bar-action' onClick={onReply}> reply </a>
          ]
        </span>
      )}
    </div>
  )
}

ActionBar.defaultProps = {
  editMode: false,
}

ActionBar.propTypes = {
  id: PropTypes.string,
  editTitle: PropTypes.string,
  editBody: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  category: PropTypes.string,
  hideCategory: PropTypes.bool,
  editMode: PropTypes.bool.isRequired,
  onEdit: PropTypes.func,
  onSave: PropTypes.func,
  onDelete: PropTypes.func,
  onCancel: PropTypes.func,
  onReply: PropTypes.func,
}

export default ActionBar
