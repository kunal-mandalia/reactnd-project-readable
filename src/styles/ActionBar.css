.action-bar {
  font-size: small;
  color: rgb(140, 140, 140);
  margin-top: 4px;
}

.action-bar-separator {
  border: none;
}

.action-bar-action:hover {
  cursor: pointer;
  color: #222222;
}