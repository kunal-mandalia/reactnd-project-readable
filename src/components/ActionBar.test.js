import React from 'react'
import ActionBar from './ActionBar'
import renderer from 'react-test-renderer'

const props = {
  date: Date.now(),
  author: `Kunal Mandalia`
}

describe(`<ActionBar .../>`, () => {
  it(`should match snapshot`, () => {
    const tree = renderer.create(<ActionBar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
