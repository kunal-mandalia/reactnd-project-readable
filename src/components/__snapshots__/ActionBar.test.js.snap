// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionBar .../> should match snapshot 1`] = `
<div
  className="action-bar"
>
  <hr
    className="action-bar-separator"
  />
  <span
    className="action-bar-date"
  >
     
    now
     
  </span>
  <span
    className="action-bar-author"
  >
     by 
    Kunal Mandalia
     
  </span>
  <span
    className="action-bar-category"
  >
     in 
     
  </span>
  <span
    className="action-bar-actions"
  >
    [
    <a
      className="action-bar-action"
      onClick={[Function]}
    >
       edit 
    </a>
    ]
  </span>
</div>
`;
